framework/src/problems/SubProblem.C

```diff
@@ -933,18 +947,57 @@ SubProblem::addAlgebraicGhostingFunctor(std::shared_ptr<GhostingFunctor> algebra
   EquationSystems & eq = es();
   auto n_sys = eq.n_systems();
 
-  for (MooseIndex(n_sys) i = 0; i < n_sys; ++i)
-    eq.get_system(i).get_dof_map().add_algebraic_ghosting_functor(algebraic_gf, to_mesh);
+  eq.get_system(0).get_dof_map().add_algebraic_ghosting_functor(algebraic_gf, to_mesh);
```

Comment:
The code in `SubProblem::addAlgebraicGhostingFunctor(shared pointer, bool ) `is very similar to `SubProblem::addAlgebraicGhostingFunctor(&, bool)`. Could we make a little helper, and reuse the code?